### PR TITLE
Adding missing ions in CGM stackplots

### PR DESCRIPTION
**Pull request by jnburchett**
Previously, it was necessary to have AbsLines of a given species already associated for CGMAbsSys.stack_plot() to produce a panel showing the spectral regions covering that species' transitions.  This adds that functionality.

### CODE REVIEW

File: pyigm/cgm/cgm.py [hunk 1]
```diff
@@ -323,10 +330,26 @@ def stack_plot(self,to_plot,pvlim=None,maxtrans=3,return_fig=True,**kwargs):
                 else: # Pick components in system closest to z_cgm
                     thesecomps = pu.get_components(self,tp)
                     if len(thesecomps)==0:
-                        continue
-                    # Find component with redshift closest to systemic
-                    compvels = np.array([np.median(tc.vlim.value) for tc in thesecomps])
-                    comp = thesecomps[np.argmin(np.abs(compvels))]
+                        if add_missing_lines is True:
+                            if isinstance(tp,str):
```
Comment (profxj) [inline pyigm/cgm/cgm.py]: I'm not sure str is 2.7 compatible, but
that's why we are on the way to 3.X

File: pyigm/cgm/cgm.py [hunk 2]
```diff
@@ -323,10 +330,26 @@ def stack_plot(self,to_plot,pvlim=None,maxtrans=3,return_fig=True,**kwargs):
...
+                                tup = ltai.name_to_ion(tp)
+                            else:
+                                tup = tp
+                            comp = AbsComponent(self.coord,tup,zcomp=self.z,
+                                                vlim=[-100.,100.]*u.km/u.s)
+                            comp.add_abslines_from_linelist()
+                            if spec is not None:
```
Comment (profxj) [inline pyigm/cgm/cgm.py:342]: Maybe add a comment here.  I can't remember why the
first element is enough for setting ```spec```

File: pyigm/cgm/cgm.py [hunk 3]
```diff
@@ -335,35 +358,53 @@ def stack_plot(self,to_plot,pvlim=None,maxtrans=3,return_fig=True,**kwargs):
...
-                        complines = np.array(complines) # For the indexing
-                        try:
-                            tokeep = [complines[compnames == sn][0] for sn in strong['name']]
-                            lines2plot.extend(tokeep)
-                        except:
-                            warnings.warn('{} covered by spectra but not in'
-                                          'components list'.format(sn))
+                        for i,sn in enumerate(strong['name']):
+                            if sn in compnames:
+                                tokeep = [complines[compnames == sn][0]]
```
Comment (profxj) [inline pyigm/cgm/cgm.py]: Add a few comments in here if you can.
Thanks!

**Review by profxj — COMMENTED** [review]
A few minor requests.
Merge when you are happy.

File: pyigm/cgm/cgm.py (hunk 1 above)
Comment (jnburchett) [reply pyigm/cgm/cgm.py]: It actually works under 2.7 on my desktop machine!

File: pyigm/cgm/cgm.py (hunk 2 above)
Comment (jnburchett) [reply pyigm/cgm/cgm.py:342]: Good catch!  It isn't enough.

File: pyigm/cgm/cgm.py (hunk 3 above)
Comment (jnburchett) [reply pyigm/cgm/cgm.py]: Done.

**Comment by jnburchett** [comment]
merging

**Pull request merged**